pkg/aws/eni/node.go

```diff
@@ -279,8 +288,21 @@ func (n *Node) allocateENI(s *types.Subnet, a *allocatableResources) error {
 	scopedLog = scopedLog.WithField(fieldEniID, eniID)
 	scopedLog.Info("Created new ENI")
 
+	attachRetries := 0
+retryAttachment:
 	attachmentID, err := n.manager.ec2API.AttachNetworkInterface(index, nodeResource.Spec.ENI.InstanceID, eniID)
 	if err != nil {
+		// Handle the situation if the index is already in use, this
+		// can happen if the local list of ENIs is oudated.
+		// Retry the attachment to avoid having to delete the ENI
+		if strings.Contains(err.Error(), "already has an interface attached at device") {
```

Comment:
just a suggestion,
comparing errors with strings.Contains feels a bit clumsy. [ec2API says in docs](https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#EC2.AttachNetworkInterface):
`Use runtime type assertions with awserr.Error's Code and Message methods to get detailed information about the error.`
```suggestion
		if err, ok := err.(awserr.Error); ok && (err.Code() == "InvalidNetworkInterface.InUse" || err.Code() == "InvalidParameterValue") {
```

Additional check for `InvalidParameterValue` is here because doc for 
[InvalidNetworkInterface.InUse](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/errors-overview.html) says:
`The specified interface is currently in use and cannot be deleted or attached to another instance. Ensure that you have detached the network interface first. If a network interface is in use, you may also receive the InvalidParameterValue error.`

Reply:
The error code returned is `InvalidParameterValue` but it's used for a wide set of invalid parameters.

So we'll still need a check like this:
```
if err, ok := err.(awserr.Error); ok && err.Code() == "InvalidParameterValue" && strings.Contains(err.Message(), "interface attached at device") {
```